exporter/azuremonitorexporter/metricexporter.go

```diff
@@ -0,0 +1,63 @@
+// Copyright OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azuremonitorexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/azuremonitorexporter"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+type metricExporter struct {
+	config           *Config
+	transportChannel transportChannel
+	logger           *zap.Logger
+}
+
+func (exporter *metricExporter) onMetricData(context context.Context, metricData pmetric.Metrics) error {
+	resourceMetrics := metricData.ResourceMetrics()
+	metricPacker := newMetricPacker(exporter.logger)
```

Comment:
I think maybe it is better to let `metricPacker` be stored as a field of metricExporter and initialized only once in `newMetricExporter` ?

If the metricPacker is initialized in onMetricData, I think it will be initialized for many times.

Reply:
Agree